### PR TITLE
Improve Azure Driver

### PR DESCRIPTION
**Pull request by lborguetti**
Guys, 

This PR improves Azure Driver allowing to customize options like resource group name, location, vm size, image reference in the molecule configuration file. 

Other improve is create the instance with Azure Managed Disk. The current driver does not remove the storage account created for the instance disk.

Example:
```yaml
molecule/azure/molecule.yml
---
driver:
  name: azure
  location: eastus2
  resource_group_name: infra_test
platforms:
  - name: UbuntuServer16.04
    location: eastus2
    vm_size: Standard_A0
    image:
      offer: UbuntuServer
      publisher: Canonical
      sku: '16.04-LTS'
      version: latest
```

Let me know if this is useful.

Thanks,

### CODE REVIEW

**Comment by wilmardo** [comment]
I think @retr0h is busy at the moment but the PR is missing an update of the Cerberus schema validation. Take a look at #1436 for some pointers @retr0h gave me how to implement this 👍

**Comment by retr0h** [comment]
Yes, what @wilmardo said.

**Comment by retr0h** [comment]
I'm going to hold off from merging this.  As you know [Ansible is adopting Molecule](https://groups.google.com/forum/#!topic/ansible-project/ehrb6AEptzA), therefore I do not want to merge anything that may impact future architectural decisions, as I will no longer be leading the development of this project.

**Comment by gundalow** [comment]
Once the repo has been moved under Ansible we can look at getting this merged.

In the short term we welcome testing and feedback on this PR.

Thanks for your understanding.

**Comment by lborguetti** [comment]
Thanks for clarification.

**Comment by gundalow** [comment]
@lborguetti Hi, repo move is done, so I'm reviewing the PR backlog. I see from comment about from @wilmardo that "Cerberus schema validation" may need updating, could you please take a look at that?

**Comment by lborguetti** [comment]
@gundalow yes. I'll take a look. 

Thanks,

File: molecule/cookiecutter/molecule/{{cookiecutter.role_name}}/{{cookiecutter.molecule_directory}}/{{cookiecutter.scenario_name}}/molecule.yml [hunk 1]
```diff
@@ -12,6 +16,12 @@ lint:
 platforms:
 {%- if cookiecutter.driver_name == 'azure' %}
   - name: instance
+    vm_size: Standard_A0
```
Comment (lwm) [inline molecule/cookiecutter/molecule/{{cookiecutter.role_name}}/{{cookiecutter.molecule_directory}}/{{cookiecutter.scenario_name}}/molecule.yml]: Maybe this is also said in https://github.com/ansible/molecule/pull/1491#issuecomment-434651331 but I think we now need to add a schema to validate this against as in https://github.com/ansible/molecule/blob/master/molecule/model/schema_v2.py. If so, there should also be some unit tests that verify validation works as expected.

**Comment by themr0c** [comment]
close/repopren to trigger travis build.

**Comment by ssbarnea** [comment]
I would gladly support this if it passes CI and once https://github.com/ansible/molecule/issues/1840 is addressed. Until then I will mark it as blocked. The only other option to merge it is if we can find cores that can test it on their own azure accounts.

**Comment by decentral1se** [comment]
Pushing out of next release as per https://github.com/ansible/molecule/pull/1491#issuecomment-473514297. Hope we can get moving on this soon but we're lacking some resources right now. It's on the stack!

**Comment by ssbarnea** [comment]
@lborguetti can you please fix the conflicts? If you do I will make sure I test it and merge it.

**Comment by ssbarnea** [comment]
I am closing this because code changed too much since it it was created and apparently the original uploader no longer active.

If the improvements are still desired please do them in small chucks, one improvement at a time. This makes easier to review and test, less likely to have conflicts. In the end merging all improvement in small chucks would be faster than in a single one.

For example few days ago I fixed the issue of using the harcoded region.

**Pull request closed without merging**